inference-engine/src/transformations/src/transformations/common_optimizations/fq_mul_fusion.cpp

```diff
@@ -0,0 +1,107 @@
+                            ngraph::Output<ngraph::Node> out_high,
+                            ngraph::Output<ngraph::Node> mul_constant) {
+    const auto mul_out_low = std::make_shared<ngraph::opset4::Multiply>(out_low, mul_constant);
+    const auto mul_out_high = std::make_shared<ngraph::opset4::Multiply>(out_high, mul_constant);
+
+    ngraph::OutputVector folded_out_low(1), folded_out_high(1);
+
+    if (!mul_out_low->constant_fold(folded_out_low, {out_low, mul_constant})) {
+      throw ngraph::ngraph_error("Could not constant fold the output_low Multiply operation");
+    }
+
+    if (!mul_out_high->constant_fold(folded_out_high, {out_high, mul_constant})) {
+      throw ngraph::ngraph_error("Could not constant fold the output_high Multiply operation");
+    }
+
+    return {folded_out_low[0], folded_out_high[0]};
+  }
+} // namespace
+
+// This transformation multiplies the "output_low" and "output_high" inputs of the FQ operation
+// by the constant value that before transormation is used to multiply the output of FQ.
+// The Multiply node is removed along with the constant (C6).
+//
+//            C1 C2 C3 C4 C5
+//            |  |  |  |  |
+//            |  |  |  |  |                     C1 C2 C3   C4*C6    C5*C6
+//            v  v  v  v  v                     |  |  |      |        |
+//          +--------------+                    |  |  |      |        |
+//          | FakeQuantize |                    v  v  v      v        v
+//          +--------------+                  +--------------------------+
+//                 |                =====>    |       FakeQuantize       |
+//                 v                          +--------------------------+
+//            +----------+                                  |
+//            | Multiply | <--- C6                          v
+//            +----+-----+
+//                 |
+//                 v
+//
+
+ngraph::pass::FakeQuantizeMulFusion::FakeQuantizeMulFusion() {
+  const auto fq_output_low_p = ngraph::pattern::wrap_type<opset4::Constant>();
+  const auto fq_output_high_p = ngraph::pattern::wrap_type<opset4::Constant>();
+
+  const auto fq_node_p = ngraph::pattern::wrap_type<opset4::FakeQuantize>(
+      {ngraph::pattern::any_input(),
+       ngraph::pattern::wrap_type<opset4::Constant>(),
+       ngraph::pattern::wrap_type<opset4::Constant>(),
+       fq_output_low_p,
+       fq_output_high_p},
+      pattern::consumers_count(1));
+
+  const auto mul_constant_p = ngraph::pattern::wrap_type<opset4::Constant>();
+  const auto mul_node_p = ngraph::pattern::wrap_type<opset4::Multiply>(
+      {fq_node_p, mul_constant_p}, pattern::consumers_count(1));
+
+  ngraph::matcher_pass_callback callback = [=](pattern::Matcher &m) {
+    const auto &pattern_map = m.get_pattern_value_map();
+
+    const auto fq_node = std::dynamic_pointer_cast<ngraph::opset4::FakeQuantize>(
+        pattern_map.at(fq_node_p).get_node_shared_ptr());
+    if (!fq_node) {
```

Comment:
If you don't use any specific fields from FQ class you don't have to cast node to FQ type.